cmd/invoker/main.go

```diff
@@ -0,0 +1,228 @@
+	var startNum int
+	_, err := fmt.Sscanf(parts[len(parts)-1], "%d", &startNum)
+	if err != nil {
+		return "", 0, fmt.Errorf("invalid number suffix in username: %v", err)
+	}
+
+	return prefix, startNum, nil
+}
+
+// invokeSync performs a synchronous Lambda invocation
+func invokeSync(client *lambda.Client, functionName string, event LambdaEvent, outputFile string) {
+	fmt.Printf("Invoking Lambda function for %s...\n", event.Username)
+
+	// Convert event to JSON
+	payload, err := json.Marshal(event)
+	if err != nil {
+		fmt.Printf("Error marshaling event: %v\n", err)
+		return
+	}
+
+	// Invoke Lambda function
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(event.Delay)*time.Millisecond+5*time.Minute)
+	defer cancel()
+	result, err := client.Invoke(ctx, &lambda.InvokeInput{
+		FunctionName: aws.String(functionName),
+		Payload:      payload,
+	})
+	if err != nil {
+		fmt.Printf("Error invoking Lambda function: %v\n", err)
+		return
+	}
+
+	// Check for Lambda execution errors
+	if result.FunctionError != nil {
+		fmt.Printf("Lambda execution error for %s: %s, %s\n", event.Username, *result.FunctionError, string(result.Payload))
+		return
+	}
+
+	// Parse the Lambda response
+	var response LambdaResponse
+	if err := json.Unmarshal(result.Payload, &response); err != nil {
+		fmt.Printf("Error parsing Lambda response: %v\n", err)
+		return
+	}
+
+	// Parse the response body
+	var responseBody ResponseBody
+	if err := json.Unmarshal([]byte(response.Body), &responseBody); err != nil {
+		fmt.Printf("Error parsing response body (%q): %v\n", response.Body, err)
+		return
+	}
+
+	// Print the response
+	if response.StatusCode == http.StatusOK {
+		fmt.Printf("[%s] Message: %s\n", event.Username, responseBody.Message)
+	} else {
+		fmt.Printf("[%s] Error: %s\n", event.Username, responseBody.Error)
+	}
+
+	if outputFile != "" && responseBody.Screenshot != "" {
+		fmt.Printf("Saving screenshot to %s\n", outputFile)
+		// Decode base64 screenshot and save to file
+		err := saveScreenshot(outputFile, responseBody.Screenshot)
+		if err != nil {
+			fmt.Printf("Error saving screenshot: %v\n", err)
+		}
+	}
+}
+
+// invokeConcurrent performs multiple concurrent Lambda invocations
+func invokeConcurrent(client *lambda.Client, functionName string, baseEvent LambdaEvent, count int, outputFileTemplate string) {
+	fmt.Printf("Invoking %d Lambda functions concurrently...\n", count)
+
+	// Parse username to extract prefix and starting number
+	prefix, startNum, err := parseUsername(baseEvent.Username)
+	if err != nil {
+		fmt.Printf("Error parsing username: %v\n", err)
+		return
+	}
+
+	var wg sync.WaitGroup
+
+	// Launch goroutines for each invocation
+	for i := 0; i < count; i++ {
+		wg.Add(1)
+		go func(index int, event LambdaEvent) {
+			defer wg.Done()
+
+			// Add random sleep proportional to count
+			// The higher the count, the more spread out the starts will be
+			maxSleepMs := count * 200 // 200ms per count as a scaling factor
```

Comment:
Do you see this as something configurable down the line?

Reply:
Yeah definitely. I left it like this for now, but down the line we'll need to do something better.